internal/ledgerutil/compare.go

```diff
@@ -23,48 +23,67 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	resultsDir      = "ledger_comparison_results"
+	allDivsFilename = "all_divergences.json"
+)
+
 // Compare - Compares two ledger snapshots and outputs the differences in snapshot records
-// This function will overwrite the file at outputPath if it already exists
-func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count int, err error) {
+// This function will not overwrite existing files
+func Compare(snapshotDir1 string, snapshotDir2 string, outputDir string) (count int, allRes string, err error) {
 	// Check the hashes between two files
 	hashPath1 := filepath.Join(snapshotDir1, kvledger.SnapshotSignableMetadataFileName)
 	hashPath2 := filepath.Join(snapshotDir2, kvledger.SnapshotSignableMetadataFileName)
 
 	equal, err := snapshotsComparable(hashPath1, hashPath2)
 	if err != nil {
-		return 0, err
+		return 0, "", err
 	}
 
 	if equal {
-		return 0, errors.New("both snapshots public state hashes are same. Aborting compare")
+		return 0, "", errors.New("both snapshots public state hashes are same. Aborting compare")
+	}
+
+	// Create the output files
+	err = os.Chdir(outputDir)
+	if err != nil {
+		return 0, "", err
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		return 0, "", err
+	}
+	err = os.Mkdir(resultsDir, 0o777)
+	if err != nil {
+		return 0, "", err
 	}
 
-	// Create the output file
-	jsonOutputFile, err := newJSONFileWriter(outputFile)
+	allRes = filepath.Join(wd, resultsDir, allDivsFilename)
+	jsonOutputFile, err := newJSONFileWriter(allRes)
```

Comment:
I find this code a bit ad-hoc. Earlier, the file name was being passed to this function. All we need to change is that this function will take the outputdir (instead of file) and make the file name a constant in this file.

Reply:
The code is implemented this way to mimic the way the cd command works on unix based shells, the goal being to improve usability. Otherwise, the user would have to specifically use the full file path for consistent results.

Reply:
Not sure I understand what you mean by "user would have to specifically use the full file path for consistent results". Isn't outDir already contains the path, either full or relative, whatever user wants?